python/cudf/cudf/core/indexing.py

```diff
@@ -266,11 +266,41 @@ def _getitem_tuple_arg(self, arg):
             else:
                 return columns_df.index._get_row_major(columns_df, arg[0])
         else:
-            df = DataFrame()
-            for col in columns_df.columns:
-                # need Series() in case a scalar is returned
-                df[col] = Series(columns_df[col].loc[arg[0]])
-            df.columns = columns_df.columns
+            if isinstance(arg[0], slice):
+                start_index, stop_index = columns_df.index.find_label_range(
+                    arg[0].start, arg[0].stop
+                )
```

Comment:
May need to assert a stepsize of 1 here or handle a non (1, None) stepsize.

Reply:
We handle the stepsize in _slice, so why not use it ?

Reply:
If we handle it in `_slice` then that's perfect.